recipes/_install.rb

```diff
@@ -55,23 +55,39 @@ def nsp
   only_if(&manifest_missing)
 end
 
-package node['splunk']['package']['base_name'] do
-  source splunk_file
-  version "#{node['splunk']['package']['version']}-#{node['splunk']['package']['build']}"
-  provider node['splunk']['package']['provider']
-  only_if(&manifest_missing)
-  if platform_family?('windows')
+if platform_family? 'rhel', 'fedora', 'amazon'
```

Comment:
I get why windows is separate, but is there a reason we can't just use the package resource and let chef determine the proper provider to use automatically for the others?

Reply:
Because these aren't the automatically selected providers. On [EL Chef selects yum](https://github.com/chef/chef/blob/v14.6.35/lib/chef/provider/package/yum.rb#L40) (which might be ok... not entirely sure, we were manually picking the RPM provider earlier), on Debian flavors [chef selects apt-get](https://github.com/chef/chef/blob/v14.6.35/lib/chef/provider/package/apt.rb#L28) (which [cannot use a local source](https://github.com/chef/chef/blob/v14.6.35/lib/chef/provider/package/apt.rb#L45-L48) )